atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -205,42 +229,33 @@ private void debugLogStateOfPool() {
 
     private void checkAndUpdateBlacklist() {
         // Check blacklist and re-integrate or continue to wait as necessary
-        for (Map.Entry<InetSocketAddress, Long> blacklistedEntry : blacklistedHosts.entrySet()) {
+        for (Map.Entry<InetSocketAddress, Long> blacklistedEntry : unresponsiveHostBlacklist.getHosts().entrySet()) {
             long backoffTimeMillis = TimeUnit.SECONDS.toMillis(config.unresponsiveHostBackoffTimeSeconds());
             if (blacklistedEntry.getValue() + backoffTimeMillis < System.currentTimeMillis()) {
                 InetSocketAddress host = blacklistedEntry.getKey();
                 if (isHostHealthy(host)) {
-                    blacklistedHosts.remove(host);
+                    unresponsiveHostBlacklist.removeHost(host);
                     log.error("Added host {} back into the pool after a waiting period and successful health check.", host);
                 }
             }
         }
     }
 
-    private void addToBlacklist(InetSocketAddress badHost) {
-        blacklistedHosts.put(badHost, System.currentTimeMillis());
-    }
-
     private boolean isHostHealthy(InetSocketAddress host) {
-        CassandraClientPoolingContainer testingContainer = null;
         try {
-            testingContainer = new CassandraClientPoolingContainer(host, config);
-            testingContainer.runWithPooledResource(describeRing);
+            currentPools.get(host).runWithPooledResource(describeRing);
```

Comment:
I started working on making currentPools thread-safe but it's a pretty ugly change. Are you planning on looking at it? It might make sense to just disallow config refreshes until we have time to make it thread-safe